lib/appium.js

```diff
@@ -405,25 +402,33 @@ class AppiumDriver extends BaseDriver {
     };
   }
 
-  async attachUnexpectedShutdownHandler (driver, innerSessionId) {
-    // Remove the session on unexpected shutdown, so that we are in a position
-    // to open another session later on.
-    // TODO: this should be removed and replaced by a onShutdown callback.
-    try {
-      await driver.onUnexpectedShutdown; // this is a cancellable promise
-      // if we get here, we've had an unexpected shutdown, so error
-      throw new Error('Unexpected shutdown');
-    } catch (e) {
-      if (e instanceof B.CancellationError) {
-        // if we cancelled the unexpected shutdown promise, that means we
-        // no longer care about it, and can safely ignore it
-        return;
-      }
-      log.warn(`Closing session, cause was '${e.message}'`);
-      log.info(`Removing session ${innerSessionId} from our master session list`);
-      await sessionsListGuard.acquire(AppiumDriver.name, () => {
-        delete this.sessions[innerSessionId];
-      });
+  attachUnexpectedShutdownHandler (driver, innerSessionId) {
+    const removeSessionFromMasterList = (cause) => {
+      log.warn(`Closing session, cause was '${(cause || {}).message}'`);
```

Comment:
Since this is the only use of `cause`, the default can be in the signature
```js
const removeSessionFromMasterList = (cause = {}) => {
```

Reply:
changed